src/NHibernate/Cfg/Loquacious/ICacheConfiguration.cs

```diff
@@ -17,6 +20,18 @@ public interface ICacheConfigurationProperties
 		string RegionsPrefix { set; }
 		int DefaultExpiration { set; }
 		void Provider<TProvider>() where TProvider : ICacheProvider;
-		void QueryCache<TFactory>() where TFactory : IQueryCacheFactory;
+		[Obsolete("This method is invalid and should not be used. Use the QueryCacheFactory extension method instead.", true)]
+		void QueryCache<TFactory>() where TFactory : IQueryCache;
+	}
+
+	// 6.0 TODO: merge into ICacheConfigurationProperties
+	public static class CacheConfigurationPropertiesExtensions
+	{
+		public static void QueryCacheFactory<TFactory>(this ICacheConfigurationProperties config) where TFactory : IQueryCacheFactory
+		{
+			ReflectHelper
+				.CastOrThrow<CacheConfigurationProperties>(config, "Setting the query cache factory with Loquacious")
+				.QueryCacheFactory<TFactory>();
```

Comment:
@InlineAsm, that is how we handle simulating addition of an interface method: only the NHibernate implementation will be supported, so long for external implementations. This is a valid stance for things not expected to be implemented outside of NHibernate itself, which is the case for the loquacious configuration. (It is likely not even injectable with custom implementations anyway.)

(For cases which are expected to be implemented outside of NHibernate, we additionally introduce for the cast a brand new interface till the next major version, to be implemented by who wants to support the feature.)